### ISSUE TITLE
slow search pane

### ISSUE THREAD

**Issue by birbilis** [seajax/seajax#23]
Maybe I have too many fields marked as searchable in my .cxml and lots of content, but search box could probably be made faster.

probably search directly as you type instead of searching in the background - maybe could use background workers in newer browsers or see if search is taking more than some msec and cancel it, with a timeout set at every char pressed (and resetting countdown at next one if not fired yet) so that if you type and pause your typing speed a bit it would then search (but not search if you type too fast or try searching and abort if some max time has elapsed [this is separate from the timeout to search if user paused or slowed down typing]) 

or maybe else the search algorithm or structures could be optimized

**Comment by sethbrenith** [comment]
Agreed, there's probably a lot of room for improvement here. The search box is using a pretty simple implementation, because it happened to be fast enough on the collections I was testing and I was much more focused on the perf of zoom/pan/animate scenarios.
